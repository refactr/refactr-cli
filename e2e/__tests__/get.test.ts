import { executeWithStdoutOnly } from '../helpers/execute';

describe('refactrctl get', () => {
  test('throws on missing subcommand', async () => {
    await expect(executeWithStdoutOnly(['get'])).rejects.toMatchSnapshot();
  });

  describe('credential', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['get', 'credential'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('job', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['get', 'job'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline-revision', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['get', 'pipeline-revision'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['get', 'pipeline'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('runner', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['get', 'runner'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('run', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['get', 'run'])
      ).rejects.toMatchSnapshot();
    });
  });
});
