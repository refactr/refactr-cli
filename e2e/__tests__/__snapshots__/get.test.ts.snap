// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactrctl get credential throws on missing arguments 1`] = `
"Usage: refactrctl <credential-id> [options]

Positionals:
  credential-id  ID of the credential to fetch  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]
      --project-id  ID of the project containing the credential  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl get job throws on missing arguments 1`] = `
"Usage: refactrctl <job-id> [options]

Positionals:
  job-id  ID of the job to fetch  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]
      --project-id  ID of the project containing the job  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl get pipeline throws on missing arguments 1`] = `
"Usage: refactrctl get <command> [options]

Commands:
  refactrctl get credential <credential-id>      Get credential details
  refactrctl get job <job-id>                    Get job details
  refactrctl get project <project-id>            Get project details
  refactrctl get organization <organization-id>  Get organization details
  refactrctl get run <run-id>                    Get run details
  refactrctl get pipeline-revision <revision>    Get pipeline-revision details

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]

Unknown argument: pipeline
"
`;

exports[`refactrctl get pipeline-revision throws on missing arguments 1`] = `
"Usage: refactrctl <revision> [options]

Positionals:
  revision  Revision number of the pipeline revision to fetch  [number] [required]

Options:
      --version      Show version number  [boolean]
  -v, --verbose      Print detailed output  [boolean]
      --format       Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address      Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token   Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help         Show help  [boolean]
      --filter       Filter output using JsonPath  [string]
      --project-id   ID of the project containing the pipeline revision  [string] [required]
      --pipeline-id  ID of the pipeline for the pipeline revision  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl get run throws on missing arguments 1`] = `
"Usage: refactrctl <run-id> [options]

Positionals:
  run-id  ID of the run to fetch  [string] [required]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]
      --project-id  ID of the project containing the run  [string] [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl get runner throws on missing arguments 1`] = `
"Usage: refactrctl get <command> [options]

Commands:
  refactrctl get credential <credential-id>      Get credential details
  refactrctl get job <job-id>                    Get job details
  refactrctl get project <project-id>            Get project details
  refactrctl get organization <organization-id>  Get organization details
  refactrctl get run <run-id>                    Get run details
  refactrctl get pipeline-revision <revision>    Get pipeline-revision details

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]

Unknown argument: runner
"
`;

exports[`refactrctl get throws on missing subcommand 1`] = `
"Usage: refactrctl get <command> [options]

Commands:
  refactrctl get credential <credential-id>      Get credential details
  refactrctl get job <job-id>                    Get job details
  refactrctl get project <project-id>            Get project details
  refactrctl get organization <organization-id>  Get organization details
  refactrctl get run <run-id>                    Get run details
  refactrctl get pipeline-revision <revision>    Get pipeline-revision details

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --filter      Filter output using JsonPath  [string]

Command must be specified
"
`;
